llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -581,9 +593,14 @@ static bool canProduceS20(const MachineRegisterInfo &MRI,
   case TargetOpcode::G_CONSTANT:
   case TargetOpcode::G_IMPLICIT_DEF:
     return true;
-  default:
+  default: {
+    // Since generic extract vector element opcodes are determined at runtime,
+    // we call a function in if statement to handle them.
```

Comment:
Nit: they are not determined at run-time, it's just that they are different for each AIE version, so we need an extra layer to abstract them.